data.py

```diff
@@ -0,0 +1,83 @@
+import csv
+import json
+import operator
+
+import nlu
+import questions
+
+
+class Similarity(object):
+    def __init__(self, intent: str, score: float):
+        self.intent = intent
+        self.score = score
+
+    def __repr__(self):
+        return '<Similarity {intent} {score}>'.format(
+            intent=self.intent, score=self.score)
+
+
+class SimilarityResult(object):
+
+    def __init__(self, query_text: str, intents: [Similarity]):
+        self.query = query_text
+        self.intents = intents
+
+    def __repr__(self):
+        return '<SimilarityResult query {query} topScoringIntent {topScoringIntent} intents {intents}>'.format(
+            query=self.query, topScoringIntent=self.topScoringIntent, intents=self.intents)
+
+    @property
+    def intents(self):
+        return self._intents
+
+    @intents.setter
+    def intents(self, intents: [Similarity]):
+        self._intents = intents
+        self.topScoringIntent = max(
+            self.intents, key=operator.attrgetter('score'))
```

Comment:
I don't like this approach; remember, `'score'` is not `.score`, the difference is you can't get any hint from a string, but you could have one originally, at here we have a common way is:
```py
max(list, key=lambda item: item.score)
```